platform-bukkit/src/main/java/net/kyori/adventure/platform/bukkit/MinecraftReflection.java

```diff
@@ -141,6 +142,65 @@ public static boolean hasClass(final @NotNull String... classNames) {
     return null;
   }
 
+  /**
+   * Search a handle for a class method.
+   *
+   * @param holderClass a class
+   * @param methodModifiers method modifiers
+   * @param methodName a method name
+   * @param returnClass a method return class
+   * @param parameterClasses an array of method parameter classes
+   * @return a method handle or {@code null} if not found
+   */
+  public static MethodHandle searchMethod(final @Nullable Class<?> holderClass, final @Nullable int[] methodModifiers, final String methodName, final @Nullable Class<?> returnClass, final Class<?>... parameterClasses) {
+    return searchMethod(holderClass, methodModifiers, new String[] {methodName}, returnClass, parameterClasses);
+  }
+
+  /**
+   * Search a handle for a class method.
+   *
+   * @param holderClass a class
+   * @param methodModifiers method modifiers
+   * @param methodNames a method names
+   * @param returnClass a method return class
+   * @param parameterClasses an array of method parameter classes
+   * @return a method handle or {@code null} if not found
+   */
+  public static MethodHandle searchMethod(final @Nullable Class<?> holderClass, final @Nullable int[] methodModifiers, final @Nullable String@NotNull[] methodNames, final @Nullable Class<?> returnClass, final Class<?>... parameterClasses) {
+    if (holderClass == null || returnClass == null) return null;
+    for (final Class<?> parameterClass : parameterClasses) {
+      if (parameterClass == null) return null;
+    }
+
+    for (final String methodName : methodNames) {
+      if (methodName == null) continue;
+      try {
+        return LOOKUP.findVirtual(holderClass, methodName, MethodType.methodType(returnClass, parameterClasses));
+      } catch (final NoSuchMethodException | IllegalAccessException e) {
+      }
+    }
+
+    for (final Method method : holderClass.getDeclaredMethods()) {
+      if (!matchModifiers(method.getModifiers(), methodModifiers)
+              || !Arrays.equals(method.getParameterTypes(), parameterClasses)) continue;
+      try {
+        return LOOKUP.findVirtual(holderClass, method.getName(), MethodType.methodType(returnClass, parameterClasses));
+      } catch (final NoSuchMethodException | IllegalAccessException e) {
+      }
+    }
+    return null;
+  }
+
+  private static boolean matchModifiers(final int modifier, final @Nullable int[] modifiers) {
```

Comment:
Is there a specific reason why `methodModifiers` is an array? Modifiers are typically represented as bitset in one int.

Reply:
Agreed. It's simple to verify whether a bit set contains any of a modified set of items: `bitSet & (FLAG_1 | FLAG_2) != 0`. Consider using this pattern instead.